syntax = "proto3";

package ML;

service MLService {
    rpc SubmitMLTask(TaskRequest) returns (TaskResponse){}
    rpc SubmitTest(TestRequest) returns (TestResponse){}
    rpc SubmitAggTask(AggregateRequest) returns (AggregateResponse){}
}

message TaskRequest {
    string fn = 1;
    string to_save = 2;
    int32 node_id = 3;
}

message TaskResponse {
    string result = 1;
}

message TestRequest {
    string fn = 1;
}

message TestResponse {
    float accuracy = 1;
    float loss = 2;
}

message AggregateRequest {
    repeated string fns = 1;
    string fn = 2;
}

message AggregateResponse {
    string fn = 1;
}